### PR TITLE
Junit Serialization

### PR DESCRIPTION
**Pull request by rcwoolston**
Implementing #2122 to add Junit serialization to the set of output capabilities

### CODE REVIEW

File: deepchecks/core/serialization/check_failure/junit.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Module containing junit serializer for the CheckFailure type."""
+import xml.etree.ElementTree as ET
```
Comment (matanper) [inline deepchecks/core/serialization/check_failure/junit.py:12]: looks like a new dependency, how much kb does it add?

File: tests/serialization/test_suite_result_serialization.py [hunk 2]
```diff
@@ -210,6 +211,96 @@ def test_json_serialization():
...
+    X, y = load_iris(return_X_y=True, as_frame=True)
+    X[label_col] = y
+    df_train, df_test = train_test_split(X, stratify=X[label_col], random_state=0)
+
+    # Train Model
+    rf_clf = RandomForestClassifier(random_state=0)
+    rf_clf.fit(df_train.drop(label_col, axis=1), df_train[label_col])
+
+    ds_train = Dataset(df_train, label=label_col, cat_features=[])
+    ds_test = Dataset(df_test, label=label_col, cat_features=[])
```
Comment (matanper) [inline tests/serialization/test_suite_result_serialization.py:280]:
```suggestion
def test_junit_serialization_with_real_data(iris_split_dataset_and_model):
    ds_train, ds_test, rf_clf = iris_split_dataset_and_model
```

**Review by matanper — COMMENTED** [review]
Overall looking good 👍 
need to add `xml` to the requirements file

File: deepchecks/core/serialization/check_failure/junit.py (hunk 1 above)
Comment (rcwoolston) [reply deepchecks/core/serialization/check_failure/junit.py:12]: It should be part of the standard library already.

**Comment by rcwoolston** [comment]
I am going to need to rebase, I forgot to setup my email  in git on this machine and create a new PR

**Comment by matanper** [comment]
@rcwoolston Write to us when you are ready 👍

**Comment by rcwoolston** [comment]
Closing, will open a new one with the fixed author line

**Pull request closed without merging**